sdks/python/apache_beam/runners/portability/stager.py

```diff
@@ -526,6 +534,14 @@ def _create_jar_packages(jar_packages, temp_dir):
 
     return resources
 
+  @staticmethod
+  def _should_download_remote_extra_packages(package):
```

Comment:
I think maybe the best policy is to just never download remote packages here. If the user wanted to download packages, I figure they could just do it themselves and pass them in as local files. WDYT?

Reply:
I agree we can remove it, but I was hesitant because I wasn't sure if that would break existing use cases (i.e. non HTTP downloads) although it seems to be low risk

Reply:
Yeah, I think it is better to make a small breaking change here, since otherwise we are left with inconsistent behavior. We can document the change in the release notes, and there is a fairly easy workaround.

Reply:
Done.